app/lib/screens/wallets/wallet_info.dart

```diff
@@ -270,7 +277,70 @@ class _WalletDetailsWidgetState extends ConsumerState<WalletDetailsWidget> {
                     ),
                   ),
                 ),
-              )
+              ),
+            const SizedBox(height: 40),
+            Text(
+              'KYC Verification',
+              style: Theme.of(context).textTheme.titleLarge!.copyWith(
+                    color: Theme.of(context).colorScheme.onSurface,
+                  ),
+            ),
+            Center(
+              child: SizedBox(
+                width: MediaQuery.of(context).size.width - 40,
+                child: ElevatedButton(
+                  onPressed: () async {
+                    if (wallet.verificationStatus != 'VERIFIED') {
+                      await termsAndConditionsDialog(
+                          context: context, wallet: wallet);
+
+                      final completer = Completer<void>();
+                      StreamSubscription? subscription;
+                      try {
+                        subscription =
+                            Events().onEvent(IdentityCallbackEvent, (event) {
+                          if (event is IdentityCallbackEvent &&
+                              event.type == 'success') {
+                            logger.i(
+                                '[Event Listener] IdentityCallbackEvent with success received.');
+                            completer.complete();
+                          }
+                        });
+
+                        await completer.future;
+                        await walletsRef.verifyWallet(wallet.name);
+                      } catch (e) {
+                        logger.e(
+                            '[Event Listener] Error while waiting for event: $e');
+                      } finally {
+                        if (subscription != null) {
+                          await subscription.cancel();
+                        }
+                      }
+                    } else {
+                      showIdentityDetails(context, wallet.tfchainSecret);
+                    }
+                  },
+                  style: ElevatedButton.styleFrom(
+                    backgroundColor: wallet.verificationStatus != 'VERIFIED'
+                        ? Theme.of(context).colorScheme.errorContainer
+                        : Theme.of(context).colorScheme.primaryContainer,
+                  ),
+                  child: Text(
+                    wallet.verificationStatus != 'VERIFIED'

```

Comment:
please use the `VerificationStatus` enum
```suggestion
                    wallet.verificationStatus != VerificationState.VERIFIED.name

```

Reply:
please apply that everywhere